src/Foundatio.Repositories/Options/CacheOptions.cs

```diff
@@ -63,12 +62,14 @@ public static class SetCacheOptionsExtensions
             return options;
         }
 
-        public static T CacheExpiresAt<T>(this T options, DateTime? expiresAtUtc) where T : ICommandOptions
+        public static T CacheExpiresAt<T>(this T options, DateTime? expiresAtUtc, TimeProvider timeProvider = null) where T : ICommandOptions
         {
+            timeProvider ??= TimeProvider.System;
```

Comment:
can time provider be set on options? and then we remove this?

Reply:
Good idea. That will let us be able to set it from the configuration automatically.